xlsxwriter/worksheet.py

```diff
@@ -52,6 +62,59 @@
 )
 from .xmlwriter import XMLwriter
 
+if TYPE_CHECKING:
+    from typing_extensions import Concatenate, ParamSpec, Protocol, overload
+
+    ReturnTypeT_co = TypeVar("ReturnTypeT_co", covariant=True)
```

Comment:
I would prefer this to be `ReturnTypeT` without the `_co` unless there is some strong convention around adding `_co`. But either way it isn't going to be clear what it means to an end user.